src/ontime/model/model.py

```diff
@@ -7,33 +8,40 @@
 from .libs.skforecast.forecaster_autoreg import (
     ForecasterAutoreg as SkForecastForecasterAutoreg,
 )
+from .libs.skforecast.forecaster_autoreg_multi_variate import (
+    ForecasterAutoregMultiVariate as SkForecastForecasterAutoregMultiSeries,
+)
 
 
 class Model(AbstractBaseModel):
     """
-    Generic wrapper around all implemented time series libraries
+    Generic wrapper around SkForecast and DartsForecast time series libraries
+
+    The model is automatically selected based on the size of the time series.
+    It is chosen once and then kept for the whole lifecycle of the model.
     """
 
     def __init__(self, model, **params):
         super().__init__()
-        if isinstance(model, ModelMeta):
-            # Darts Models
-            self.model = DartsForecastingModel(model, **params)
-        else:
-            # scikit-learn API compatible models
-            self.model = SkForecastForecasterAutoreg(model, **params)
+        self.model = model
+        self.params = params
+        self.is_model_undefined = True
 
-    def fit(self, ts, **params):
+    def fit(self, ts: TimeSeries, **params) -> "Model":
         """
         Fit the model to the given time series
         :param ts: TimeSeries
         :param params: Parameters to pass to the model
         :return: self
         """
+        if self.is_model_undefined:
+            self._set_model(ts)
+            self.is_model_undefined = False
+
         self.model.fit(ts, **params)
         return self
```

Comment:
Implementation choose: First fit define the model than, we let model fit make the job and define if we can refit or not. This way, we are not taking "risk" and let libraries implementations resolve this question. It may have different behaviors (See if a convention/unique behavior should be implemented)